src/main/java/org/folio/qm/service/RecordService.java

```diff
@@ -0,0 +1,129 @@
+package org.folio.qm.service;
+
+import static org.folio.qm.util.JsonUtils.objectToJsonString;
+
+import io.vertx.core.json.JsonObject;
+import java.util.Date;
+import java.util.Objects;
+import java.util.UUID;
+import lombok.extern.log4j.Log4j2;
+import org.folio.AdditionalInfo;
+import org.folio.ParsedRecord;
+import org.folio.RawRecord;
+import org.folio.Record;
+import org.folio.processing.mapping.defaultmapper.RecordMapper;
+import org.folio.processing.mapping.defaultmapper.RecordMapperBuilder;
+import org.folio.qm.client.SourceStorageClient;
+import org.folio.qm.client.model.ParsedRecordDto;
+import org.folio.qm.client.model.RecordTypeEnum;
+import org.folio.qm.mapper.ExternalIdsHolderMapper;
+import org.folio.qm.service.support.MappingMetadataProvider;
+import org.folio.qm.util.ErrorUtils;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.context.request.async.DeferredResult;
+
+@Log4j2
+public abstract class RecordService<T> {
+
+  private final MappingMetadataProvider mappingMetadataProvider;
+  private final SourceStorageClient sourceStorageClient;
+  private final ExternalIdsHolderMapper externalIdsHolderMapper;
+
+  protected RecordService(MappingMetadataProvider mappingMetadataProvider,
+                          SourceStorageClient sourceStorageClient,
+                          ExternalIdsHolderMapper externalIdsHolderMapper) {
+    this.mappingMetadataProvider = mappingMetadataProvider;
+    this.sourceStorageClient = sourceStorageClient;
+    this.externalIdsHolderMapper = externalIdsHolderMapper;
+  }
+
+  public abstract RecordTypeEnum supportedType();
+
+  public abstract void update(UUID parsedRecordId,
+                              DeferredResult<ResponseEntity<Void>> updateResult,
+                              ParsedRecordDto parsedRecordDto);
+
+  protected T getMappedRecord(ParsedRecordDto parsedRecordDto, String recordType, String mapperName) {
```

Comment:
We should ommit usage of ParsedRecordDto, it will be removed after implementation.